pkg/controller/config.go

```diff
@@ -170,6 +180,25 @@ func configForwardfor(conf *types.HAProxyConfig) {
 	}
 }
 
+func (cfg *haConfig) createDNSResolvers() {
+	resolvers := dnsresolvers.ParseDNSResolvers(cfg.haproxyConfig.DNSResolvers)
+	for _, backend := range cfg.ingress.Backends {
+		if _, ok := backend.DNSResolvers.DNSResolvers[backend.DNSResolvers.UseResolver]; !ok {
+			if resolverData, ok := resolvers[backend.DNSResolvers.UseResolver]; ok {
+				backend.DNSResolvers.DNSResolvers[backend.DNSResolvers.UseResolver] = resolverData
+			} else if backend.DNSResolvers.UseResolver != "" {
+				backend.DNSResolvers.UseResolver = ""
+			}
+		}
+		for resolverName, resolverData := range backend.DNSResolvers.DNSResolvers {
+			if _, ok := resolvers[resolverName]; !ok {
+				resolvers[resolverName] = resolverData
+			}
+		}
+	}
+	cfg.DNSResolvers = resolvers
+}
```

Comment:
* Resolvers are global lists, annotations have precedence over configmap, so why not simply iterate backends and override the default resolvers without a single `if`?
* Warn about "resolver name %v not found"
* What if two annotations declare the same resolver? This need to be at least warned - or resolvers from annotations could have a prefix so colisions would never happen

Reply:
- With prefixes we would get multiple (same) data in haproxy config file.
- I will override configmap setup with one from annotation without checking if it exist or not
- warning 'not found' can and will be added, will still must check `if` resolver actually exist or not
- I'm not sure what do you mean if two annotations declare same resolver. You can only define it in configmap or with annotations.

Reply:
I can use annotations to declare new resolvers on more than one ingress resource.

Reply:
Don't know if I understood. Ingress have name and namespace, and both configurations are unique. If name and namespace match, you have the same ingress and you have the same resolver annotation.

What I usually do is: immutable configurations or behavior are command-line options, global or infra/ops configurations are configmap options, useful configurations to devs are annotations. dns resolvers are global and sounds to me like a infra/ops responsibility so perhaps it fits better on configmap only.